install/scripts/craycli_init.py

```diff
@@ -0,0 +1,753 @@
+            'credentials': [
+                {
+                    'type': 'password',
+                    'value': password,
+                },
+            ],
+            'groups': [group,]
+        }
+
+        # make the call and parse response
+        response = self._kc_master_admin_client.post(createUserUrl, json=req_body)
+        if response.status_code == 409:
+            LOGGER.debug("User %r already exists", username)
+            return
+        response.raise_for_status()
+
+        # for craycli to work, the user must be a member of a group with
+        # the correct permissions
+        userId = self.get_cli_user_id(username)
+        if userId == "":
+            LOGGER.error(f"Unable to get keycloak user: {username} - user creation failed")
+            sys.exit(1)
+        groupId = self.get_grp_id(group)
+        if groupId == "":
+            LOGGER.error(f"Unable to get keycloak group id for group: {group} - user creation failed")
+            sys.exit(1)
+
+        # put together the required url to add the user to the group
+        addGrpUrl = (f"{self.keycloak_base}/admin/realms/{self.SHASTA_REALM_NAME}"
+                    f"/users/{userId}/groups/{groupId}")
+        response = self._kc_master_admin_client.put(addGrpUrl)
+        response.raise_for_status()
+
+        LOGGER.info(f"Created user {username} in group {group}")
+
+    # delete the given user from keycloak
+    def delete_user(self, username):
+        # check the input
+        if username == "":
+            LOGGER.error("Attempting to delete empty username")
+            return
+
+        # get the user id of the desired user
+        userId = self.get_cli_user_id(username)
+        if userId == "":
+            LOGGER.debug(f"Specified user {username} does not exist in keycloak")
+            return
+
+        # user exists, so delete this user from keycloak
+        url = f"{self.keycloak_base}/admin/realms/{self.SHASTA_REALM_NAME}/users/{userId}"
+        response = self._kc_master_admin_client.delete(url)
+        response.raise_for_status()
+        LOGGER.info("Deleted user %r", username)
+
+    # get the keycloak user id for a given username
+    def get_cli_user_id(self, username):
+        LOGGER.debug(f"Looking for id of user: {username}")
+        # check the input
+        if username == "":
+            LOGGER.error("Attempting to find the user id of an empty username")
+            return ""
+
+        # search for the users that match the expected user
+        url = (f"{self.keycloak_base}/admin/realms"
+               f"/{self.SHASTA_REALM_NAME}/users/?username={username}")
+        response = self._kc_master_admin_client.get(url)
+        response.raise_for_status()
+
+        # parse through the responses - returns a list of json objects
+        user_id = ""
+        items = response.json()
+        for item in items:
+            if item["username"] == username:
+                user_id = item["id"]
```

Comment:
You could put in a 'break' after this line to break out of the for loop once you've found the user_id.